kcalcore/icaltimezones.cpp

```diff
@@ -1313,10 +1313,10 @@ QList<QDateTime> ICalTimeZoneSourcePrivate::parsePhase( icalcomponent *c,
   }
 
   // Convert DTSTART to QDateTime, and from local time to UTC
-  const QDateTime localStart = toQDateTime( dtstart );   // local time
   dtstart.second -= prevOffset;
   dtstart.is_utc = 1;
   const QDateTime utcStart = toQDateTime( icaltime_normalize( dtstart ) );   // UTC
+  const QDateTime localStart = utcStart.toLocalTime();  // local time
```

Comment:
Interesting... testTwo() succeeds (along with testOne() and testThree()) if we have the second patch but not the first patch included.  So it looks like the first patch is not necessary, now.

However, if we put a line like:
    qWarning() << "calculating transitions within interval:" << klocalStart.toString() << "->" << maxTime.toString();
just above the line:
    const DateTimeList dts = r.timesInInterval( klocalStart, maxTime );
then you can see without the first patch, it prints:
    calculating transitions within interval: "" -> "2035-05-21T00:00:00"
whereas with the first patch included, it prints:
    calculating transitions within interval: "1981-03-29T03:00:00" -> "2035-05-21T00:00:00"

Nonetheless, given the potential for that first patch to cause other issues, perhaps we should not do that one, and just apply the second patch in this PR?